crates/client/src/lib.rs

```diff
@@ -304,6 +306,12 @@ impl<R: RegistryStorage, C: ContentStorage> Client<R, C> {
             map_root = checkpoint.as_ref().map_root
         );
 
+        let operator = self
+            .registry
+            .load_operator()
+            .await?
+            .map(|op| op.state.head().as_ref().map(|h| h.digest.clone()).unwrap());
```

Comment:
I think this should be:

```rust
let operator = self
    .registry
    .load_operator()
    .await?
    .unwrap_or_default();
```

The reason being is that we want to use the same operator validation state we might have stored previously when validating new records for the operator log.

And then where we use it in `FetchRequest`:

```rust
operator: operator.state.head().as_ref().map(|h| h.digest.clone())
```

Note that it's valid for this to be `None` if we created a default operator info above (i.e. we haven't stored any state, so start from the beginning).

Finally, where we do the operator log validation, we should updating `operator` and not creating a new operator information every time; there should also be a single save of the operator info.